docs/architecture/adr-016-protocol-versions.md

```diff
@@ -188,8 +183,10 @@ message ResponseInfo {
 }
 ```
 
-The existing `version` fields should be called `software_version` but we leave
-them for now to reduce the number of breaking changes.
+The existing `version` field should be called `software_version` but we will leave it for now to reduce the number of breaking changes.
+Upon handshake, the node will check that the returned version string matches `TMCoreSemVer` else it will return an error. 
```

Comment:
should we add a sentence that it should match the major version? Currently it is `v0.34.0-rc4-120-gbaa20a4b9` but compatibility should be focused on `v0.34`